tap_woo/client.py

```diff
@@ -45,87 +47,31 @@ def authenticator(self) -> BasicAuthenticator:
             password=self.config.get("consumer_secret", ""),
         )
 
-    @property
-    def http_headers(self) -> dict:
-        """Return the http headers needed.
-
-        Returns:
-            A dictionary of HTTP headers.
-        """
-        headers = {}
-        if "user_agent" in self.config:
-            headers["User-Agent"] = self.config.get("user_agent")
-        # If not using an authenticator, you may also provide inline auth headers:
-        # headers["Private-Token"] = self.config.get("auth_token")  # noqa: ERA001
-        return headers
-
-    def get_new_paginator(self) -> BaseAPIPaginator:
-        """Create a new pagination helper instance.
-
-        If the source API can make use of the `next_page_token_jsonpath`
-        attribute, or it contains a `X-Next-Page` header in the response
-        then you can remove this method.
-
-        If you need custom pagination that uses page numbers, "next" links, or
-        other approaches, please read the guide: https://sdk.meltano.com/en/v0.25.0/guides/pagination-classes.html.
-
-        Returns:
-            A pagination helper instance.
-        """
-        return super().get_new_paginator()
-
-    def parse_response(self, response: requests.Response) -> Iterable[dict]:
-        """Parse the response and return an iterator of result records.
-
-        Args:
-            response: The HTTP ``requests.Response`` object.
-
-        Yields:
-            Each record from the source.
-        """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
-    
     def get_url_params(self, context, next_page_token):
-        self.logger.info(f"Next page token: {next_page_token}")
+        self.logger.debug(f"Next page token: {next_page_token}")
         params = {
             "per_page": 100,
-            "order": "asc",
-            }
-
+            "order": "asc"
+        }
         starting_date = self.get_starting_timestamp(context)
         if starting_date:
-            params["modified_after"] = starting_date
+            params["modified_after"] = starting_date.strftime(DATETIME_FORMAT)
+
+        end_date = self.config.get("end_date")
+        if end_date:
+            params["modified_before"] = pendulum.parse(end_date).strftime(DATETIME_FORMAT)
 
         if next_page_token is not None:
-            params["page"] = next_page_token
+            params["page"] = dict(parse_qsl(next_page_token.query)).get('page')
```

Comment:
It feels very weird to pass around the full `next` page url as `next_page_token`, and then parse out the next page value inside of get_url_params (as opposed to parsing it out in the paginator) - but i see that this is exactly the pattern recommended in the docs: https://sdk.meltano.com/en/latest/guides/pagination-classes.html#migrate-your-rest-tap-to-use-the-new-pagination-classes.

I also see (in separate testing) that when requesting the last available page, a `next` url is no longer returned.  Neat!

Reply:
I didn't like this approach too much because since we have the next page we should use the full url directly without the need to get the page and update the params.

But maybe I can open a PR and update it Meltano sdk core.